flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -226,6 +227,9 @@ public SchedulerBase(
                         completedCheckpointStore,
                         checkpointsCleaner,
                         checkpointIdCounter,
+                        new CheckpointStatsTracker(
```

Comment:
Wondering if the `CheckpointStatsTracker` dependency should come from the constructor as other dependencies.

Reply:
> You mean adding the parameter to the SchedulerBase constructor?

Yes, that's what I meant. 

I didn't realize that `CheckpointStatsTracker` is not passed as an interface. If this were the case, it would make sense to me to be able to use a different implementation, e.g., for testing purposes. Let's leave it as that.